### PR TITLE
Update Qt5 recipes

### PR DESCRIPTION
**Pull request by sbourdeauducq**
@ccordoba12 
@msarahan

If you need anything else, feel free to take it from there: https://github.com/m-labs/conda-recipes

### CODE REVIEW

**Comment by ccordoba12** [comment]
These are my initial comments:

1. Qt 5.6 was just released, so please update the recipes to compile that version instead.
2. Your Windows builds (as present in anaconda.org) don't have QtWekit nor QtWebEngineWidgets compiled. We need one of them (preferably the last one) for Spyder, so please add it to the recipes here.
3. There's no bld.bat for icu, but that's a dependency for Windows too, so please add it.
4. After pull request #616 most of your changes the qt5 and pyqt5 recipes for Linux and Mac are not necessary. I'll add comments to the PR to let you know what needs to be reverted.

File: qt5/bld.bat [hunk 1]
```diff
@@ -1,46 +1,64 @@
+:: change qt.conf name for qt4 coexistence
+patch -p1 <%RECIPE_DIR%/qt5_conf.patch
+
```
Comment (ccordoba12) [inline qt5/bld.bat]: We don't need this patch after PR #616. Please remove it.

Comment (ccordoba12) [inline qt5/bld.bat]: Sorry, sorry, we need it for Windows. It's ok to leave it :-)

File: qt5/build.sh [hunk 2]
```diff
@@ -1,96 +1,93 @@
 #!/bin/bash
 
-BIN=$PREFIX/lib/qt5/bin
-QTCONF=$BIN/qt.conf
-
-chmod +x configure
+# change qt.conf name for qt4 coexistence
+patch -p1 <$RECIPE_DIR/qt5_conf.patch
```
Comment (ccordoba12) [inline qt5/build.sh]: Now, we don't need this patch for Linux and Mac, please remove it.

File: qt5/build.sh [hunk 3]
```diff
@@ -1,96 +1,93 @@
...
+        -release \
+        -shared \
+        -nomake examples \
+        -nomake tests \
+        -skip enginio \
+        -qt-pcre \
+        -system-xkbcommon-x11 \
+        -no-gtkstyle \
+        -dbus \
+        -no-gstreamer
```
Comment (ccordoba12) [inline qt5/build.sh]: Please leave the same options as before.

Comment (ccordoba12) [inline qt5/build.sh]: I'm pretty confident with the list of options we're passing to configure. There are things we really need, like `-gtkstyle` (for Ubuntu), and `-qt-xcb` (to not depend on the xcb version present in the system).

So please revert to the previous list of options.

File: qt5/build.sh [hunk 4]
```diff
@@ -1,96 +1,93 @@
...
+        -qt-pcre \
+        -no-c++11 \
+        -no-framework \
+        -no-dbus \
+        -no-mtdev \
+        -no-harfbuzz \
+        -no-xinput2 \
+        -no-xcb-xlib \
+        -no-libudev \
+        -no-egl
```
Comment (ccordoba12) [inline qt5/build.sh]: What options did you exactly changed and/or added here? Please leave them in the same order as before and your new ones at the end so I can easily what you've changed :-)

File: qt5/build.sh [hunk 5]
```diff
@@ -1,96 +1,93 @@
...
-Prefix = $PREFIX/lib/qt5
-Libraries = $PREFIX/lib
-Headers = $PREFIX/include/qt5
-
-EOF
+cp $RECIPE_DIR/qt5_unix.conf $PREFIX/bin/qt5.conf
+echo "Prefix = .." >>$PREFIX/bin/qt5.conf
 
+cp $RECIPE_DIR/qt5_unix.conf $PREFIX/lib/qt5/bin/qt5.conf
+echo "Prefix = ../../.." >>$PREFIX/lib/qt5/bin/qt5.conf
```
Comment (ccordoba12) [inline qt5/build.sh]: There's no need to have this copies here. Please revert to what we had previously, I know what's needed to be done, and I'll do it after this PR is merged.

File: qt5/meta.yaml [hunk 6]
```diff
@@ -1,46 +1,41 @@
...
-    - libpng        1.6*     [unix]
-    - jpeg                   [unix]
-    - zlib                   [unix]
-    - bison                  [linux]
-    - flex                   [linux]
+    - freetype               [unix]
     - fontconfig             [linux]
-    - gst-plugins-base       [linux]
-    - libxcb                 [linux]
+    # also needed on [linux]: apt-get install flex bison gperf libx11-dev libxext-dev libxfixes-dev libxi-dev libxrender-dev libxcb1-dev libx11-xcb-dev libxcb-glx0-dev libxcb-keysyms1-dev libxcb-image0-dev libxcb-shm0-dev libxcb-icccm4-dev libxcb-sync0-dev libxcb-xfixes0-dev libxcb-shape0-dev libxcb-randr0-dev libxcb-render-util0-dev libxkbcommon-dev libxkbcommon-x11-dev
```
Comment (ccordoba12) [inline qt5/meta.yaml]: Please remove this huge comment and move it to Notes.md

File: qt5/meta.yaml [hunk 7]
```diff
@@ -1,46 +1,41 @@
...
     - icu
-    - freetype               [unix]
-    - zlib                   [unix]
     - openssl                [unix]
-    - libpng        1.6*     [unix]
-    - jpeg                   [unix]
+    - freetype               [unix]
     - fontconfig             [linux]
-    - gst-plugins-base       [linux]
-    - libxcb                 [linux]
```
Comment (ccordoba12) [inline qt5/meta.yaml]: Please leave this list as it was before.

**Comment by ccordoba12** [comment]
As I said, most of the things you changed for Linux and Mac are really not necessary.

The Windows recipes still need some tweaking to use the right Visual Studio version per Python version, but @msarahan can help you with that.

File: qt5/build.sh (hunk 2 above)
Comment (ccordoba12) [inline qt5/build.sh]: This is what PR #616 made unnecessary :-)

Comment (ccordoba12) [inline qt5/build.sh]: I mean, after Qt4 packages are released based on the recipes present in #616, Qt5 packages based on our current recipes could be installed side by side with them, so this patch is not needed.

**Comment by sbourdeauducq** [comment]
I supposed this can be closed now?

**Comment by ccordoba12** [comment]
Yep, thanks for your help. I took some ideas from your work, specially from the PyQt5 recipe.

**Comment by sbourdeauducq** [comment]
:+1: Thanks for upgrading and merging.

**Pull request closed without merging**